game/missiongenerator/convoygenerator.py

```diff
@@ -32,17 +32,46 @@ def generate(self) -> None:
                 self.generate_convoy(convoy)
 
     def generate_convoy(self, convoy: Convoy) -> VehicleGroup:
-        group = self._create_mixed_unit_group(
-            convoy.name,
-            convoy.route_start,
-            convoy.units,
-            convoy.player_owned,
-        )
-        group.add_waypoint(
-            convoy.route_end,
-            speed=kph(40).kph,
-            move_formation=PointAction.OnRoad,
-        )
+        if self.game.settings.perf_moving_convoys:
+            group = self._create_mixed_unit_group(
+                convoy.name,
+                convoy.route_start,
+                convoy.units,
+                convoy.player_owned,
+            )
+
+            group.add_waypoint(
+                convoy.route_end,
+                speed=kph(40).kph,
+                move_formation=PointAction.OnRoad,
+            )
+
+        else:
+            # perf_moving_convoys is disabled, so add the convoy between the route start and route end.
+            # This option aims to remove long routes for ground vehicles between control points,
+            # since the CPU load for long routes on DCS is pretty heavy.
+            # Put a some distance between the convoy and its first waypoint, so it'll move a little
+            # before stopping and possibly find a road to drive on.
+            start_x = (3 * convoy.route_start.x + convoy.route_end.x) / 4
+            start_y = (3 * convoy.route_start.y + convoy.route_end.y) / 4
```

Comment:
What if this is water, or a building?

I don't think this can reasonably be done without input from the campaign designer to specify more than one spawn point (https://github.com/dcs-liberation/dcs_liberation/issues/1002). Until then I'm okay with the option, but I think it still needs to spawn in the normal location instead of at an arbitrary location that might be inside a building, forest, or lake.

Reply:
The waypoints are now generated from the campaign definition supply routes by utilizing Frontline. Here the direct route from Gelendzhik to Novorossiysk would have placed the convoy in the sea with the previous implementation: 
![image](https://user-images.githubusercontent.com/89945461/142253404-c3462200-6b72-4a62-b105-53b52ccb6d25.png)

The convoy is now correctly placed on land.